ietf-optical-impairment-topology.yang

```diff
@@ -952,9 +1147,178 @@ module ietf-optical-impairment-topology {
       "This augment is only valid for optical impairment with sliceable
       transponder model";
    }
-   description 
-     "Tunnel termination point augmentation for sliceable transponder
-     model.";
-   uses sliceable-transponder-attributes; 
+   description
+     "Tunnel termination point augmentation for sliceable transponder model.";
+   uses sliceable-transponder-attributes;
   }
+
+  augment "/nw:networks/nw:network/nw:node/tet:te"
+        + "/tet:te-node-attributes" {
+    when "/nw:networks/nw:network/nw:network-types"
+       + "/tet:te-topology/optical-imp-topo:optical-impairment-topology" {
+
+      description
+        "This augment is only valid for Optical Impairment topology";
+    }
+    description
+      "node attributes augmentantion for optical-impairment RAODM node";
+
+    list path-impairment {
+      key "path-impairment-id";
+      config false;
+      description "list of set of optical impairments related to ROADM ";
+
+      leaf path-impairment-id {
+        type uint32; 
+        description "index of the path-impairment list";
+      }
+      choice impairment-type {
+        description "type path impairment";
+        case roadm-express-path {
+          uses roadm-express-path; 
+        }  
+        case roadm-add-path {
+          uses roadm-add-path; 
+        }          
+        case roadm-drop-path {
+          uses roadm-drop-path; 
+        }
+      }
+    } // list path impairments 
+  } // augmentation for optical-impairment ROADM 
+
+  augment "/nw:networks/nw:network/nw:node/tet:te/"
+        + "tet:information-source-entry/tet:connectivity-matrices" {
+    when "/nw:networks/nw:network/nw:network-types"
+       + "/tet:te-topology/optical-imp-topo:optical-impairment-topology" {
+      description
+        "This augment is only valid for Optical Impairment topology ";
+    }  
+    
+    description
+      "Augment default TE node connectivity matrix information source.";
+
+    leaf path-impairment {
+      type leafref {
+        path "../../../tet:te-node-attributes/path-impairment/path-impairment-id";
+      }
+      description "pointer to the list set of ROADM optical impairments";
+    }

```

Comment:
Since this augment is for the whole matrix, is this only for express path ? if the matrix is only for express connectivity I would suggest to rename the leaf with express-path-impairment similarly to the namong "add-path-impairment". Else I would suggest to ad a generic name roadm-path-impairment, just to avoid any confusion with a path that would be a network path (through the whole network). I do not know if this last case is relevant.

Reply:
I think for similarity to the case of LLCL and add/drop path, the suggestion makes sense to me. 
Any other opinion ? if not I would make the modification.

Reply:
Ok for me but I would suggest the name "roadm-path-impairment" since we still do not know whether the connectivity matrix models only express paths or also add/drop paths to external OTs

Reply:
> Ok for me but I would suggest the name "roadm-path-impairment" since we still do not know whether the connectivity matrix models only express paths or also add/drop paths to external OTs
Sha
Yes, good observation. My only doubt is that also add and drop path are ROADM-path-impairment. Shall we use this for every pointers ? (also add and drop) . Not clear to me the proposal.